go/test/endtoend/backuptransform/backup_transform_test.go

```diff
@@ -0,0 +1,205 @@
+	// every file is starting with 'header'
+	validateManifestFile(t, backupLocation)
+
+	// restore replica2 from backup, should not give any error
+	// Note: we don't need to pass in the backup_storage_transform parameter,
+	// as it is read from the MANIFEST.
+	replica2.MysqlctlProcess.ExtraArgs = []string{
+		"-db-credentials-file", dbCredentialFile}
+	// clear replica2
+	replica2.MysqlctlProcess.Stop()
+	os.RemoveAll(replica2.VttabletProcess.Directory)
+
+	// start replica2 from backup
+	err = replica2.MysqlctlProcess.Start()
+	require.Nil(t, err)
+	err = localCluster.VtctlclientProcess.InitTablet(replica2, cell, keyspaceName, hostname, shardName)
+	assert.Nil(t, err)
+	replica2.VttabletProcess.CreateDB(keyspaceName)
+	replica2.VttabletProcess.ExtraArgs = []string{
+		"-db-credentials-file", dbCredentialFile,
+		"-restore_from_backup",
+		"-backup_storage_implementation", "file",
+		"-file_backup_storage_root", localCluster.VtctldProcess.FileBackupStorageRoot}
+	replica2.VttabletProcess.ServingStatus = ""
+	err = replica2.VttabletProcess.Setup()
+	require.Nil(t, err)
+	err = replica2.VttabletProcess.WaitForTabletTypesForTimeout([]string{"SERVING"}, 25*time.Second)
+	require.Nil(t, err)
+	defer replica2.VttabletProcess.TearDown()
+
+	// validate that semi-sync is enabled for replica, disable for rdOnly
+	if replica2.Type == "replica" {
+		verifyReplicationStatus(t, replica2, "ON")
+	} else if replica2.Type == "rdonly" {
+		verifyReplicationStatus(t, replica2, "OFF")
+	}
+
+	// validate that new slave has all the data
+	cluster.VerifyRowsInTablet(t, replica2, keyspaceName, 2)
+
+	// Remove all backups
+	for _, backup := range listBackups(t) {
+		err := localCluster.VtctlclientProcess.ExecuteCommand("RemoveBackup", shardKsName, backup)
+		assert.Nil(t, err)
+	}
+
+}
+
+// TestBackupTransformErr validate backup with test_backup_error
+// backup_storage_hook, which should fail.
+func TestBackupTransformErr(t *testing.T) {
+	// restart the replica with transform param
+	err := replica1.VttabletProcess.TearDown()
+	require.Nil(t, err)
+
+	replica1.VttabletProcess.ExtraArgs = []string{
+		"-db-credentials-file", dbCredentialFile,
+		"-backup_storage_hook", "test_backup_error",
+		"-restore_from_backup",
+		"-backup_storage_implementation", "file",
+		"-file_backup_storage_root", localCluster.VtctldProcess.FileBackupStorageRoot}
+	replica1.VttabletProcess.ServingStatus = "SERVING"
+	err = replica1.VttabletProcess.Setup()
+	assert.Nil(t, err)
+
+	// create backup, it should fail
+	out, err := localCluster.VtctlclientProcess.ExecuteCommandWithOutput("Backup", replica1.Alias)
+	require.NotNil(t, err)
+	assert.Containsf(t, out, "backup is not usable, aborting it", "unexpected error received %v", err)
+
+	// validate there is no backup left
+	backups := listBackups(t)
+	assert.Equalf(t, 0, len(backups), "invalid backups: %v", backups)
+}
+
+// listBackups fetches the list of backup
+func listBackups(t *testing.T) []string {
```

Comment:
We must be using the same thing in other backup and xtrabackup tests. Can we put this in one place and re-use it?